File: 2023-2024/sourcecode/mod2/Lab7/ViewModelBasicsJava/app/src/main/java/com/example/viewmodelbasicsjava/viewmodel/WordViewModel.java
package com.example.viewmodelbasicsjava.viewmodel;

import androidx.lifecycle.ViewModel;

import com.example.viewmodelbasicsjava.data.DataProvider;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class WordViewModel extends ViewModel {
    private List<String> wordsList = new ArrayList<>();

    public List<String> getWordList() {
        return wordsList;
    }

    public WordViewModel() {
        reinitialize();
    }

    public void addWord(String word) {
        wordsList.add(word);
        Collections.sort(wordsList);
    }

    public void reinitialize() {
        wordsList.clear();
        wordsList.addAll(DataProvider.allWordsList);
        Collections.sort(wordsList);
    }

    public void clear() {
        wordsList.clear();
    }
}

File: 2023-2024/sourcecode/mod3/lab8/ViewModelBasicsKotlin/app/src/main/java/com/example/viewmodelbasics_kotlin/viewmodel/WordViewModel.kt
package com.example.viewmodelbasics_kotlin.viewmodel

import androidx.lifecycle.ViewModel
import com.example.viewmodelbasics_kotlin.data.DataProvider

class WordViewModel : ViewModel() {
    private var _wordsList: MutableList<String> = mutableListOf()
    val wordList: List<String>
        get() = _wordsList

    init {
        reinitialize()
    }

    fun addWord(word: String){
        _wordsList.add(word)
        _wordsList.sort()
    }

    fun reinitialize(){
        _wordsList.clear()
        _wordsList.addAll(DataProvider.words)
        _wordsList.sort()
    }

    fun clear(){
        _wordsList.clear()
    }
}